### DISCUSSION TITLE
No communication through tunnel interface on balena

### DISCUSSION THREAD

**Discussion by Timple** [eProsima/Fast-DDS#3951] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Both my laptop, an ubuntu server as the balena device are connected to the same zerotier network.
(Balena setup like [this](https://github.com/synapzlu/balena-zerotier)).

On my laptop I run a dds listener (using a ros2 node):
```
docker run --rm -it --net=host ros:humble /bin/bash -c 'apt-get update -qq && apt-get install -qqy ros-humble-demo-nodes-cpp && source /opt/ros/humble/setup.bash && ros2 run demo_nodes_cpp listener'
```

On the server I can run a talker:

```
docker run --rm -it --net=host ros:humble /bin/bash -c 'apt-get update -qq && apt-get install -qqy ros-humble-demo-nodes-cpp && source /opt/ros/humble/setup.bash && ros2 run demo_nodes_cpp talker'
```

This works, I am now seeing: 
```
[INFO] [1697045332.110629381] [listener]: I heard: [Hello World: 5]
```

### Current behavior

However now I try the same in balena:
```
balena ssh uuid.local
balena run --rm -it --net=host ros:humble /bin/bash -c 'apt-get update -qq && apt-get install -qqy ros-humble-demo-nodes-cpp && source /opt/ros/humble/setup.bash && ros2 run demo_nodes_cpp talker'
```
No output shown by the listener.

I intentionally dressed it down to this, typically we use docker-compose files.

When I try the same strategy with all three devices on the same local network, everything works fine.

### Steps to reproduce

See above

### Fast DDS version/commit

fastdds in docker: `ros:humble`. However same results for iron.

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

UDPv4

### Additional context

Using the Fast DDS monitor and a discovery-server. I can see the zerotier interface is not listed as communication candidate.

I tried to setup a ddsrouter with udp or tcp communication:
- no communication to the remote balena device
- yes communication to the remote ubuntu server

### XML configuration file

```xml
No XML configuration. Everything ROS Humble defaults.
```

### Relevant log output

```shell
root@f4c56ba:/# ip link show ztwdjbk2e5
21: ztwdjbk2e5: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 2800 qdisc fq_codel state UNKNOWN mode DEFAULT group default qlen 1000
    link/ether aa:8b:b7:ab:3d:42 brd ff:ff:ff:ff:ff:ff

From the docker. Same on the remote server as the balena device
```

### Network traffic capture

As expected, wireshark on my laptop does not show any messages from the balena device. 

More hints on how to debug this are appreciated. I wonder why the node on the balena device does not see the zerotier tunnel device as valid option to communicate with.

**Comment by JLBuenoLopez** [comment]
@Timple 

Regretfully, Fast DDS does not have official support for balena devices. Consequently, according to the [project's CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md) I am moving the issue to the unofficial support forum. You might also be interested on contacting [eProsima's commercial support team](https://github.com/eProsima/Fast-DDS/tree/master#getting-help) for getting support for your specific use case.

**Comment by Timple** [comment]
You are right, it belongs here. I never thought of using the discussions page as a forum, so I thought the link just reverted back to the same repository. Thanks for the move!
